### DISCUSSION TITLE
Performance Tuning

### DISCUSSION THREAD

**Discussion by jedihugo** [geldata/gel#3182] [Q&A] [answered]
In postgresql we may have and can tune stuffs like shared_mem, WAL, workable_mem, etc. Is EdgeDB auto-tuning based on EdgeQL query workload?

**Comment by elprans** [comment, marked as answer]
EdgeDB does not do any auto-tuning (at lest not yet) outside of picking `max_connections` based on system RAM.  You can tune certain PostgreSQL parameters with `CONFIGURE`, see [the documentation](https://www.edgedb.com/docs/stdlib/cfg#resource-usage).  For example:

```
CONFIGURE INSTANCE SET query_work_mem := <cfg::memory>'8MiB';
```

However, if you run EdgeDB over a standalone PostgreSQL cluster (via `--backend-dsn`), it is best to configure the underlying PostgreSQL instance directly,
